code/modules/rykka_puzzle_wips/reactor_defense.dm

```diff
@@ -0,0 +1,795 @@
+	var/verification_time = 90 SECONDS				// How long are we allowing users to have during the verification step?
+	var/verification_timeout = null					// When does verification timeout/fail? This is set in start_verification()
+	var/key_difficulty = 2							// How hard of a word/phrase do we require users to type in. Ranges from 1 (very easy) to 5 (very hard).
+	var/warmup_dangerous = FALSE					// Are our warmups dangerous (IE, do we spawn mobs during warmup phases?) FALSE by default.
+	var/threshold_enabled = FALSE					// Is the requirement to keep the reactor above x hp enabled?
+	var/damage_threshold = 0						// This is configurable, allowing you to choose if you want players to keep the reactor above x amount of HP. Formula checks health < damage_threshold, so 1500 threshold means once reactor hits 1499 HP you fail.
+	var/continuous_defense = FALSE					// Are we immediately going to start the next warmup/wave without giving explorers a break? If FALSE, this will kick the reactor back to IDLE and require explorers to hit the button to start the next wave.
+	
+	
+	// TGUI Variables, don't fuck with these unless you know what you're changing.
+	// This is the current tab we're on, used for the UI/Menu.
+	var/current_tab = IDLE
+	
+	/* 	== Spawning Variables. == */
+
+	// Change these on the subtype of the reactor, for custom mob lists.
+	// Weighted with values (not %chance, but relative weight)
+	// Can be left value-less for all equally likely
+	// Simply uncomment each additional list per wave you want to use. If you want to add more to the list, just copy the previous wave, and make sure to remove the comma at the end of the entry.
+	var/list/wave_mobs = list(
+		list(/mob/living/simple_mob/animal/wolf, /mob/living/simple_mob/mechanical/hivebot),
+//		list(/mob/living/wave2mob, /mob/living/otherwave2mob, etc),
+//		list(/mob/living/wave3mob, /mob/living/otherwave3mob, etc),
+//		list(/mob/living/wave4mob, /mob/living/otherwave4mob, etc),
+//		list(/mob/living/wave5mob, /mob/living/otherwave5mob, etc),
+//		list(/mob/living/wave6mob, /mob/living/otherwave6mob, etc)
+	)
+
+	
+	// Settings to help mappers/coders have their mobs do what they want in this case, and control the # of mobs.
+	var/faction						// To prevent infighting if it spawns various mobs, set a faction
+	var/atmos_comp = FALSE			// TRUE will set all their survivability to be within 20% of the current air
+	var/total_spawns = -1 			// Total mob spawns, over all time, -1 for no limit
+	var/simultaneous_spawns = 6 	// Max spawned mobs active (from one landmark) at one time
+	var/delayed_spawns = FALSE		// Are we delaying our spawns?
+	var/spawn_delay = 0.5 SECONDS 	// How long is our delay?
+	
+	// Internal Use Only. (For those looking as to what this does, this is a typecast. We use this to handle setting up our mobs. If you're not sure what a typecast is, look it up or ask for help in #dev)
+	var/mob/living/simple_mob/my_mob

```

Comment:
:information_source: This appears to be unused and I'm not sure what would require a casted reference to something that is presumably intended to be temporary but is attached permanently to the main object instead of a temporary var like most type casts do after a type check.

Reply:
Removed it, you were correct in that it's unused and it was left in from a prior attempt at setting up spawning